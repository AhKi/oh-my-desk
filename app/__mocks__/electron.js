export default jest.fn();

export const webContents = {
  getAllWebContents: jest.fn(() => []),
};

export const ipcMain = {
  on: jest.fn(),
};

export const BrowserWindow = jest.fn(info => ({
  close: jest.fn(),
  loadURL: jest.fn(),
  once: jest.fn(),
  on: jest.fn(),
  setAlwaysOnTop: jest.fn(),
  show: jest.fn(),
  showInactive: jest.fn(),
  getContentBounds: jest.fn(() => ({
    x: info.x,
    y: info.y,
    height: info.height,
    width: info.width,
  })),
  webContents: {
    on: jest.fn(),
    send: jest.fn(),
    openDevTools: jest.fn(),
  },
}));

export const ipcRenderer = {
  on: jest.fn(),
  send: jest.fn(),
};

export const remote = {
  getGlobal: jest.fn(),
  getCurrentWindow: jest.fn(),
  Menu: {
    buildFromTemplate: jest.fn(() => ({
      popup: jest.fn(),
    })),
  },
  app: {
    getVersion: jest.fn(),
  },
};

export const shell = {
  openExternal: jest.fn(),
};

export const app = {
  getName: jest.fn(() => 'oh-my-desk'),
  getPath: jest.fn(arg => arg),
  getVersion: jest.fn(),
  getAppPath: jest.fn(() => 'mock-path'),
  isReady: jest.fn(),
  on: jest.fn(),
};

export const Menu = {
  buildFromTemplate: jest.fn(() => ({
    popup: jest.fn(),
  })),
  setApplicationMenu: jest.fn(),
};

export const globalShortcut = {
  register: jest.fn(),
  unregister: jest.fn(),
};
