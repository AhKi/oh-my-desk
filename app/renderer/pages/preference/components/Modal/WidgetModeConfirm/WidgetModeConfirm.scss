.WidgetModeConfirm__container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  padding: 20px;
  font-family: inherit;
}

.WidgetModeConfirm__content {
  padding: 10px 0;
  font-size: 16px;
}

.WidgetModeConfirm__btn-set {
  display: flex;
  justify-content: center;
  margin: 10px;
}
