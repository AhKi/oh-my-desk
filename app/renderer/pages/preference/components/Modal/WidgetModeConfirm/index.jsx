import React from 'react';
import PropTypes from 'prop-types';
import i18n from 'constants/i18n';
import './WidgetModeConfirm.scss';

const propTypes = {
  onChangeMode: PropTypes.func,
  onModalClose: PropTypes.func,
};
const defaultProps = {
  onChangeMode() {},
  onModalClose() {},
};

class WidgetModeConfirm extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    const { onChangeMode, onModalClose } = this.props;

    onChangeMode();
    onModalClose();
  }

  render() {
    const { onModalClose } = this.props;
    const text = i18n().preference;

    return (
      <div className="WidgetModeConfirm__container">
        <h3>{text.widgetModeConfirmTitle}</h3>
        <p className="WidgetModeConfirm__content">
          {text.widgetModeConfirmContent}
        </p>
        <div className="WidgetModeConfirm__btn-set">
          <button
            className="Btn Btn--sm Btn--gray"
            type="button"
            onClick={onModalClose}
          >
            {text.changedCancel}
          </button>
          <button
            className="Btn Btn--sm Btn--primary"
            type="button"
            onClick={this.handleChange}
          >
            {text.ok}
          </button>
        </div>
      </div>
    );
  }
}

WidgetModeConfirm.propTypes = propTypes;
WidgetModeConfirm.defaultProps = defaultProps;

export default WidgetModeConfirm;
