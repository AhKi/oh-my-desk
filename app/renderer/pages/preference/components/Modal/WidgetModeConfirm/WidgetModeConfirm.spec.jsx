import React from 'react';
import { shallow } from 'enzyme';
import { remote } from 'electron';

import WidgetModeConfirm from '.';

describe('<WidgetModeConfirm />', () => {
  remote.getGlobal = jest.fn(() => () => JSON.stringify({ status: { lang: 'English' } }));

  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<WidgetModeConfirm />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange', () => {
    const onChangeMode = jest.fn();
    const onModalClose = jest.fn();
    const wrapper = shallow(
      <WidgetModeConfirm
        onChangeMode={onChangeMode}
        onModalClose={onModalClose}
      />,
    );

    wrapper.instance().handleChange();

    expect(onChangeMode).toHaveBeenCalledTimes(1);
    expect(onChangeMode).toHaveBeenCalledWith();
    expect(onModalClose).toHaveBeenCalledTimes(1);
    expect(onModalClose).toHaveBeenCalledWith();
  });
});
