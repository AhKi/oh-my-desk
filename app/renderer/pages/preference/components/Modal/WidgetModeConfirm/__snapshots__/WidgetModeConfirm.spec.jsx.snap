// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WidgetModeConfirm /> should match to snapshot when render default 1`] = `
<div>
  <h3>
    Do you want to change the window mode?
  </h3>
  <p>
    The window you are working on might not be saved.
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Undo
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Ok
  </button>
</div>
`;
